packages/slate-react/src/components/editable.tsx

```diff
@@ -474,15 +478,15 @@ export const Editable = (props: EditableProps) => {
   const decorations = decorate([editor, []])
 
   if (
-    placeholder &&
+    (renderPlaceholder || placeholder) &&
```

Comment:
I think this should still only check for `placeholder` (since the renderer will be defaulted).

More generally, I don't think we're currently solving for the ability to define any arbitrary rendering logic in the `renderPlaceholder` override. If we were (which we could discuss in a separate issue) we'd likely want it to include the check right below this as well in the rendering layer:

```ts
editor.children.length === 1 &&
Array.from(Node.texts(editor)).length === 1 &&
Node.string(editor) === ''
```

Since without that included it wouldn't be very customizable.

Reply:
I've reverted this so it only checks for placeholder and now instead of using the placeholder variable to pass the render prop I'm passing it through similarly to `renderElement` and `renderLeaf`.

Regarding the second part of your message I'm not entirely sure I follow. Would you like me to include this or shall we deal with it in a separate change?